bluefog/common/mpi_controller.cc

```diff
@@ -555,6 +626,33 @@ void MPIController::NeighborAllreduce(std::vector<TensorTableEntry>& entries) {
   timeline_ptr->ActivityEndAll(entries);
   const void* fused_input_data = buffer_data;
 
+  const void* weighted_fused_input_data = nullptr;
+  if (first_entry.dst_weighting_enabled) {
+    // Generate weighted data fusion for sending
+    timeline_ptr->ActivityStartAll(entries, "MEMCPY_IN_WEIGHT_FUSION_BUFFER");
+    void* weight_buffer_data;
```

Comment:
Please add comments about the storage structure of WeightFusionBuffer

Reply:
Done.